### PR TITLE
Vueify header

### PR DESCRIPTION
**Pull request by sharkykh**
- Convert `partials/header.mako` to a Vue component ~(as much as possible)~
- Remove all `topmenu` arguments :tada:
- Implement mobile view notifications badge (#3921)

~I still need to add some values to the API, and hopefully remove all the Python conversions.~

### CODE REVIEW

File: themes-default/slim/views/vue-components/app-header.mako [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    from medusa import app
+%>
+<script>
+Vue.component('app-header', {
+    template: '#app-header-template',
+    data() {
+        return {
+            // Python conversions
+            <% has_emby_api_key = json.dumps(app.EMBY_APIKEY != '') %>
+            hasEmbyApiKey: ${has_emby_api_key},
```
Comment (sharkykh) [inline themes-default/slim/views/vue-components/app-header.mako]: @OmgImAlexis 
I think this the last blocker to finishing this component (and maybe moving to `.vue` file).
I can't add the API key to the endpoint (#4404), and we never decided on a solution for the passwords / API keys.

Comment (OmgImAlexis) [reply themes-default/slim/views/vue-components/app-header.mako]: We should allow `PATCH` for them for now. I'm really not okay with sending the API keys and passwords back to the client if we don't have to.

Comment (sharkykh) [reply themes-default/slim/views/vue-components/app-header.mako]: Here I need to know if the API key is set, not patch it.
So what I'm saying is, there was a discussion on Slack about how to handle situations where you need to know if it's set (excluding the config pages where we currently have the value in the HTML).
That discussion never reached its conclusion - so we need to figure it out.

I've decided to just remove the emby API key check for now, just so I can move forward with this.
Left a comment in place though.

Comment (OmgImAlexis) [reply themes-default/slim/views/vue-components/app-header.mako]: Why is the value needed? We may be able to work around this in another way.

**Comment by sharkykh** [comment]
>Why is the value needed? We may be able to work around this in another way.

It was needed to ensure the button to update Emby was only visible if you had all the settings set (`enabled`, `host` and `apikey`).
It can work without it, but we're gonna hit that issue again soon with the config pages and we'll need a solution.

File: medusa/server/web/core/base.py [hunk 2]
```diff
@@ -88,19 +87,13 @@ def __init__(self, rh, filename):
             'sbDefaultPage': app.DEFAULT_PAGE,
             'loggedIn': rh.get_current_user(),
             'sbStartTime': rh.startTime,
-            'numErrors': len(classes.ErrorViewer.errors),
-            'numWarnings': len(classes.WarningViewer.errors),
             'sbPID': str(app.PID),
```
Comment (OmgImAlexis) [inline medusa/server/web/core/base.py]: Can this be removed too?

Comment (sharkykh) [reply medusa/server/web/core/base.py]: No, it's still being used everywhere.

**Comment by sharkykh** [comment]
Waiting on #4666

**Pull request merged**